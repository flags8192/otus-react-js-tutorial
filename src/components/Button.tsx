import React from 'react';
import styled from '@emotion/styled';
import { FC } from 'react';

const Root = styled.button`
  height: 30px;
  margin: auto 20px;
`;

export interface IButton {
  title?: string;
  onClick?: () => void;
}
const Button: FC<IButton> = ({ title = 'Button', onClick }) => {
  return (
    <Root role="button" onClick={onClick}>
      {title}
    </Root>
  );
};

export default Button;
