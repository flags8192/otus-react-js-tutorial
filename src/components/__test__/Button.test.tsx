import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Button from '@src/components/Button';

describe('Button', () => {
  it('render <Button>', () => {
    const handleClick = jest.fn();
    const { asFragment } = render(<Button onClick={handleClick} title="Reset" />);
    expect(asFragment()).toMatchSnapshot();
    let button = screen.getByRole(/button/gi);
    expect(button).toBeInTheDocument();
    button = screen.getByText(/Reset/gi);
    expect(button).toBeInTheDocument();
  });

  it('click on <Button>', () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick} title="Reset" />);
    const button = screen.getByRole(/button/gi);
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
