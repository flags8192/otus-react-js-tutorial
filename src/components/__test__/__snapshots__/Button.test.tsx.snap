// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render <Button> 1`] = `
<DocumentFragment>
  <button
    class="css-tv5zn1"
    role="button"
  >
    Reset
  </button>
</DocumentFragment>
`;
